src/Model/Command/ItemsFilteringTrait.php

```diff
@@ -82,10 +82,11 @@ public function addBoost(Boost $boost): self
 
     /**
      * Set boosts. Removes all previously set rules.
+     * @param Boost[] $boosts
      */
     public function setBoosts(array $boosts): self
     {
```

Comment:
Best would also be to add something like
```php
Assertion::allIsInstanceOf($boosts, Boost::class);
```
right on beginning of the method body.

And maybe then is the `array_values(...)` unnecessary, isn't it? Or in what case did this issue occur?

Reply:
The `array_values()` call was very important when the previous application layer implemented a filter to remove certain array items by key.

The `array_values()` function then created a list again from the filtered array.